src/core/readyException.js

```diff
@@ -0,0 +1,13 @@
+define( [
+	"../core"
+], function( jQuery ) {
+
+"use strict";
+
+jQuery.readyException = function( error ) {
+	window.setTimeout( function() {
+		throw error;
+	} );
```

Comment:
> I forgot @gibson042 wanted to have this reject as well so that a more general potential unhandled rejections mechanism would catch it. 

If you wanted to catch your own ready handler's rejections you can already do `$.when($.ready).then(fn).catch(errfn)` in 3.0.  What other cases do we want to handle? I feel like 99.9% of the cases are errors and as long as we manage to dump a stack we're doing all we need to do.